Alter proj4 method

calebbraun opened the pull request:
Update mapping functions

crvernon committed:
initial

crvernon committed:
accept all coord ref types; built spatial harmonization functions; expanded IO options - Shapefiles and other; activate spatial bounding box; others

crvernon committed:
added data frame to spatial polygonsdata frame function

crvernon committed:
working - added function for full data type import to render a spatial data frame

crvernon committed:
fixed import function for shp

crvernon committed:
changed functionality to accomodate sf objects and processing capabilities

calebbraun committed:
Update documentation and add first test file

rplzzz committed:
Remove references to ggalt and sp; add sf package dependency.

Also, update manual pages.

rplzzz committed:
Fix errors found in package check.

Errors are resolved, but some warnings still remain.

calebbraun committed:
Add more consistent styling and update examples and documentation (part 1)

calebbraun committed:
Add more consistent styling and update examples and documentation (part 2)

calebbraun committed:
Add tests for looking up proj4 strings

calebbraun committed:
Add a more information in examples about fetching proj4 strings

calebbraun committed:
Fix a few formatting errors from merge

calebbraun committed:
Add TODO about removing necessity of sf package for building vignettes

calebbraun committed:
Fix bug where joining data converts from sf object to pure data.frame

calebbraun committed:
Update tests, examples, and warnings

calebbraun committed:
Add new remote repository for gdal

calebbraun committed:
Adjust Travis config to add apt repository with sudo

calebbraun committed:
Try to install newest version of GDAL with simple Travis file

calebbraun committed:
Fix errors when running R CMD check and update Travis to pull ggplot2 from github

calebbraun committed:
Fix warnings when running R CMD check and remove functionality to read mapdata from text file

calebbraun committed:
Add OS-X build to CI configuration.

crvernon committed:
Solved warning for sf::st_intersection() call stating attribute variables are assumed to be spatially constant throughout all geometries. Repaired by providing a user option for attribute-geometry-relationship designation. Either constant (default), aggregate, or identity classified as follows:  [constant] a variable that has a constant value at every location over a spatial extent; examples: soil type, climate zone, land use. [aggregate]values are summary values (aggregates) over the geometry, e.g. population density, dominant land use.  [identity]values identify the geometry: they refer to (the whole of) this and only this geometry. See https://cran.r-project.org/web/packages/sf/vignettes/sf1.html#how-attributes-relate-to-geometries for further explanation.

crvernon committed:
adding cast to sf obj from dplyr left_join for gcam_join() call due to left_join returning data.frame object instead of sf data.frame

crvernon committed:
Message for st_join suppressed:  "although coordinates are longitude/latitude, it is assumed that they are planar."  This comes from the input projection being a geographic coordinate system and not a projected one when conducting topological operations such as st_intersects used in the st_join.  if longlat appears in the proj string ("+proj=longlat +ellps=WGS84 +datum=WGS84 +no_defs") then this message will present itself due to the operation being intersect according to the source code. There is no option to quiet this.  We are getting the expected return from the operation due to harmonizing the map and bounding box projection pre-join.

crvernon committed:
Suppress Warning: In st_buffer.sfc(st_geometry(x), dist, nQuadSegs) : st_buffer does not correctly buffer longitude/latitude data, dist needs to be in decimal degrees.
This warning occurs from buffering a feature that is in a geographic coordinate system (lat/long) rather than a projection one.  This makes the buffer not be exact due to the distance being calculated in decimal degrees rather than meters or kilometers. This is fine with us since we are simply using buffer as a way of zooming to include or exclude portions of the bounding extent in this call.

calebbraun committed:
Fix bug caused by agr_type being recursively defined in filter_spatial()

calebbraun committed:
Update plot_GCAM_grid so that crs for boudning box and map data is the same

calebbraun committed:
Create rdata files for map data;
fix typo in travis.yml;
small change to examples

calebbraun committed:
added rastering for grid data, still lots of bugs

calebbraun committed:
Check for world extent before trying coordinate transform

calebbraun committed:
Added simplify function for map; call regular plot gcam instead of redoing everything in plot_gcam_grid

calebbraun committed:
Removed color formatting from plotting functions

calebbraun committed:
Added tests for plot_GCAM_grid() and simplify_mapdata()

calebbraun committed:
Updated map data files and included simplified versions

calebbraun committed:
Added check for if reprojecting gridded data makes NAs;

rplzzz commented:
I'll test this on mac OS tonight or tomorrow.

calebbraun committed:
Remove support for generic orthographic projections with orientation parameter;

Continued updating examples to match changes to package; Removed pre-defined orientation vectors.

rplzzz commented:
@calebbraun Can I ask why you deleted `gcammaptools.Rproj` and added it to the `.gitignore` list?

rplzzz commented:
@calebbraun In the examples vignette, the first map seems not to be getting the correct color palette.  It's plotting in shades of blue (presumably scaled from 0-32) instead of getting the qualitative palette it used to.

In fact, what we probably want to do is move the 32-region palette from GCAM-dashboard into this package and export it.  Then we could use that palette in displaying this example.

Edit: Looking through the rest of the vignette, this problem seems to recur.  User-specified color palettes just don't look like they're getting used.

rplzzz committed:
Restore deleted Rproj file; update pkg description, and tweak examples.

calebbraun commented:
@rplzzz I deleted `gcammaptools.Rproj` file thinking it was similar to .Rhistory and just had information about my local preferences and workspace.  I'll put it back.

calebbraun committed:
Reproject gridded data with raster package

This change fixes the problem where areas of the gridded data were missing. Additionally updated examples to include a plot of gridded data.

calebbraun committed:
Fix merge conflict

calebbraun commented:
@rplzzz In regards to the color palette we talked about removing color mappings and legend options entirely and suggesting that the user use ggplot's options instead. We would need to show examples of this still, but I think that it gives the user more flexibility.

rplzzz commented:
@calebbraun Understood, and we can still do that, but having some supports for a house style would be good.  In this case, the color palette wouldn't have to be applied automatically, but should be available for users to apply if they want it, and we should do so in the examples, since plotting regions in shades of blue isn't really useful or helpful.

rplzzz commented on R/diag_header.R:
```diff
@@ -317,7 +288,7 @@ gcam14_colors<- c("Africa" = "navajowhite3",
 
 
-rgb255 <- function(r, g, b) {rgb(r,g,b, maxColorValue=255)}
+rgb255 <- function(r, g, b) {grDevices::rgb(r,g,b, maxColorValue=255)}
```
Apropos of our earlier discussion, it looks like the 32-region color palette is already in the package, so all we have to do is add it to the example in the vignette.

rplzzz commented on tests/testthat/test_gridded.R:
```diff
@@ -0,0 +1,101 @@
+  expect_true("lat" %in% colnames(co2grid))
+})
+
+test_that("plot_GCAM_grid plots with all defaults", {
```
I suspect that none of these tests will catch errors in rendering the plots, since they don't occur until the `print.gg` method gets called.  What we probably need to do is to attempt to render these to temp files, or maybe look into the section on headless testing on Travis.  However, we can kick that can a little further down the road, for now.

rplzzz commented on vignettes/examples.Rmd:
````diff
@@ -12,110 +12,110 @@ vignette: >
 This example just plots the map data frame with the GCAM region id.  Thus, you get each region colored according to a discrete color palette.
 ```{r mp1, fig.width=6, fig.height=3}
-plot_GCAM(map_32_wo_Taiwan.fort, col = 'id', proj = eck3, colorfcn=qualPalette)
+plot_GCAM(mapdata, col = 'region_id', proj = eck3)
````
This plot doesn't actually match the description above.  Instead of a discrete color palette, you get shades of blue.

rplzzz replied:
Also, is there a reason why the `region_id` columns in the maps are of type numeric rather than integer?  Having an integer type would cause R to treat it as discrete values, instead of continuous.

calebbraun replied:
I agree that `region_id` should be of type integer, but they would still be [treated as continuous values](https://github.com/tidyverse/ggplot2/issues/1340). If I add a column to the maps for region name (or add it on the fly) it can plot the regions using the `gcam32_colors` palette. Any reason not to do that?

rplzzz replied:
That sounds like the way to go.

rplzzz reviewed (changes requested):
This looks good.  We need to change the vignette to provide a couple of examples of how to change the color mapping.  Demonstrating the region32 colors on the first plot would be good, and we should also show how to do it with continuous data, like the CO2 emissions.

rplzzz committed:
Cleanup for merge.

calebbraun committed:
Update vignette to provide color mapping examples

Change the default maps to include a region_name column, which makes using
scale_fill_manual with gcam32_colors very easy. Show examples of how to do
a few different color mappings.

calebbraun committed:
Remove color mapping functions

To allow more flexibility, do not try to take the color mapping as input
anymore. Remove functions that helped determine correct color palette or
where the legend breaks should go.

calebbraun committed:
Update documentation

rplzzz commented:
@calebbraun Great job.  This is a big improvement over the old version.
